src/OpenGL/MonoMacGameView.cs

```diff
@@ -73,7 +74,14 @@ public MonoMacGameView (CGRect frame, NSOpenGLContext context) : base(frame)
 				NSOpenGLPixelFormatAttribute.ColorSize, 24,
 				NSOpenGLPixelFormatAttribute.DepthSize, 16 };
 
-			pixelFormat = new NSOpenGLPixelFormat (attribs);
+			try {
+				pixelFormat = new NSOpenGLPixelFormat (attribs);
+			} catch (Exception) {
+				// Fails on VM because there is no hardware-acceleration
+				// https://github.com/xamarin/xamarin-macios/issues/4417
+				attribs = attribs.Skip (1).ToArray ();
```

Comment:
beside bringing LINQ code, it's not really efficient (even less since there's already allocations inside the `NSOpenGLPixelFormat ` binding code).

You can probably avoid that by
1. moving `NSOpenGLPixelFormatAttribute.Accelerated` at the end (I don't think the order matter);
2. if it fails, replace the last element with `0` (which is like `nil` and will mark the end of the list)
3. try again

Should work - without LINQ code and extraneous memory allocations :)